agent/grpc-external/services/resource/read_test.go

```diff
@@ -0,0 +1,161 @@
+package resource
+
+import (
+	context "context"
+	"testing"
+
+	"github.com/hashicorp/consul/internal/resource"
+	storage "github.com/hashicorp/consul/internal/storage"
+	pbresource "github.com/hashicorp/consul/proto-public/pbresource"
+	mock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func TestRead_TypeNotFound(t *testing.T) {
+	mockRegistry := &MockRegistry{}
+	mockRegistry.On("Resolve", mock.Anything).Return(resource.Registration{}, false)
```

Comment:
Is there a reason we can't just use the real registry here? (and in the below test cases too)

Reply:
Updated all tests to not use a mock Registry or Backend since `consistency` can now be asserted against.